src/ReportGenerator.Core/Common/EnumerableExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿#if !NETSTANDARD2_1_OR_GREATER && !NETCOREAPP_2_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Palmmedia.ReportGenerator.Core.Common
+{
+    /// <summary>
+    /// Extension methods on <see cref="IEnumerable{T}"/>.
+    /// </summary>
+    internal static class EnumerableExtensions
+    {
+        /// <summary>
+        /// Returns a new enumerable collection that contains the last
+        /// <paramref name="count"/> elements from <paramref name="source"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements in the enumerable collection.</typeparam>
+        /// <param name="source">An enumerable collection instance.</param>
+        /// <param name="count">The number of elements to take from the end of the collection.</param>
+        /// <returns>A new enumerable collection that contains the last
+        /// <paramref name="count"/> elements from <paramref name="source"/>.</returns>
+        /// <remarks>
+        /// If <paramref name="count"/> is not a positive number, this method returns an empty enumerable collection.
+        /// </remarks>
+        public static IEnumerable<T> TakeLast<T>(this IEnumerable<T> source, int count)
+        {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (count <= 0)
+            {
+                return Enumerable.Empty<T>();
+            }
+
+            // Check if we have direct access to elements
+            if (source is IList<T> list)
+            {
+                if (list.Count < count)
+                {
+                    return list;
+                }
+                else
+                {
+                    return TakeLast(list, count);
+                }
+            }
+
+            // We have to iterator over the whole source and keep count elements.
+            var queue = new Queue<T>(count);
```

Comment:
would it be that leaving this block would not be better reading and would dispense with the previous comment?

`var queue = KeepElementByCount(count, source);`
ou
`return KeepElementByCount(count, source);`

o metodo
```
public static IEnumerable<T> KeepElementByCount<T>(this IEnumerable<T> source, int count)
{
            var queue = new Queue<T>(count);
            foreach (var item in source)
            {
                if (queue.Count == count)
                {
                    queue.Dequeue();
                }

                queue.Enqueue(item);
            }
            return queue;
}
```

Reply:
I applied your suggestion.